Src/PerformanceCollector/Unit.Tests/QuickPulse/PerfLib/CategorySampleTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace Unit.Tests
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class CategorySampleTests
+    {
+        [TestMethod]
+        public void CategorySampleReadsDataCorrectly()
+        {
+            // ARRANGE
+            var dataList = new List<byte>();
+            string resourceName = "Unit.Tests.QuickPulse.PerfLib.PerfData.data";
+
+            Stream stream = null;
+            try
+            {
+                stream = typeof(CategorySampleTests).Assembly.GetManifestResourceStream(resourceName);
+
+                using (StreamReader reader = new StreamReader(stream, Encoding.UTF8))
+                {
+                    stream = null;
+
+                    dataList.AddRange(reader.ReadToEnd().Split(',').Select(byte.Parse));
+                }
+            }
+            finally
+            {
+                stream?.Dispose();
```

Comment:
You would. You would also probably prefer to see a pair of nested usings and nothing else. But alas (see above) :)